File: yggdrasil/examples/maxMsg/src/maxMsgSrv.py
from yggdrasil.interface.YggInterface import YggRpcServer


print("maxMsgSrv(P): Hello!")
rpc = YggRpcServer("maxMsgSrv", "%s", "%s")

while True:
    ret, input = rpc.rpcRecv()
    if not ret:
        break
    print("maxMsgSrv(P): rpcRecv returned %s, input %.10s..." % (ret, input[0]))
    flag = rpc.rpcSend(input[0])
    if not flag:
        raise RuntimeError('maxMsgSrv(P): Error sending reply.')

print("maxMsgSrv(P): Goodbye!")

File: yggdrasil/examples/maxMsg/src/maxMsgSrv.m


disp('maxMsgSrv(M): Hello!');

rpc = YggInterface('YggRpcServer', 'maxMsgSrv', '%s', '%s');

while (1)
  [flag, vars] = rpc.recv();
  if (~flag)
    break;
  end
  fprintf('maxMsgSrv(M): rpcRecv returned %d, input %.10s...\n', ...
	  flag, char(vars{1}));
  flag = rpc.send(vars{1});
  if (~flag)
    error('maxMsgSrv(M): Error sending reply.');
  end
end

disp('maxMsgSrv(M): Goodbye!');

